src/analyzer/analyzergain.cpp

```diff
@@ -39,18 +41,39 @@ void AnalyzerGain::cleanup() {
 bool AnalyzerGain::processSamples(const CSAMPLE* pIn, SINT count) {
     ScopedTimer t(QStringLiteral("AnalyzerGain::process()"));
 
-    SINT numFrames = count / mixxx::kAnalysisChannels;
+    SINT numFrames = count / m_channelCount;
+
+    const CSAMPLE* pGainInput = pIn;
+    CSAMPLE* pMixedChannel = nullptr;
+
+    if (m_channelCount > mixxx::kAnalysisChannels) {
+        // If we have multi channel file (a stem file), we mix all the stems
+        // together in a stereo channel
```

Comment:
If we load the track in a STEM deck, this gain should be correct. But what if we load the mixed stereo signal into a sampler deck? Do we use load 5th mixed channel of the STEM file (which might have a different gain), or do we mix the 4 stem channels at load time?

Reply:
we mix the 4 stem channels at load time, as part of the SoundSource. This done automatically depending of the request max channel count when opening the file.